# Practice Probability Interview Questions

## There are 12 red balls and 12 black balls in a box. If you draw 1 ball, then a second ball, what is the probability they are the same color?

Source: Probability lecture.



In [4]:
12 / 24 # 50% chance red or black first time

0.5

In [2]:
11/23 # 47.8% for next ball to be the same

0.4782608695652174

In [14]:
(.478 * .5) # 23.9% chance to draw 2 of the same colored balls in a row

0.239

Answer: 47.826%

## Your friend is interested in sky-diving and plans to do 500 jumps. Because of equipment failures and similar problems, the probability of death on one jump is 0.001. What is the overall probability of death under your friend's plan?

- 39%
- 50%
- 51%
- 62% 


Source: Walmart Interview Question

Answer: 39%

##  Teams A and B  are playing a game with 5 matches. A has probability p to win a match. What's the probability A wins the game at the 5th match.

Source: Google Interview Question



My Wrong Answer: p if all matches are independent of each other and unique

*Right Answer:*

In [ ]:
6 * p**3 * (1-p)**2

## Bonus: You are given 8 identical looking balls. One of them is heavier than the rest of the 7 (all the others weigh exactly the same). You a provided with a simple mechanical balance and you are restricted to only 2 uses. Find the heavier ball.

Source: State Farm Interview Question



Answer 1: I'd put all 8 balls on the balance as my first use. I'd take away the balls 1 by 1 and measure the difference in weight loss. Whichever ball had a unique difference would be the heavier ball, so on my second use I would weight that and compare it to the last ball that was left standing on the balance in my original use. In this example there is a chance that the last ball was the heavier ball though, so in that case I wouldn't even need a second use.

Answer 2: Put all 8 balls on the balance. From here I should be able to mathematically find out how much each ball should weigh by using the total weight and dividing by 7 and 8 and differencing where appropriate.

Answer 3: Put 3 balls on one side and 4 on the other. Using math we should be able to figure it out.

Let's say 7 balls weigh 1 and the 8th is 2.

If we had 4 balls weighing 1 each on one side it'd weigh 4 and the other side with 4 balls would weigh 5.

# RIGHT ANSWER

1st use: 3 and 3 on each side

2nd use: you either weigh the last 2 if the first use was exactly even, or you take 2 from the 3 that tipped the scales from the first and use process of elimination